Agents/AverageSquareMetrePriceAgent/README.md

```diff
@@ -0,0 +1,255 @@
+
+```bash
+# Java >= 11
+# Test installation
+java -version
+javac -verison
+# Install in case it is missing
+sudo apt install openjdk-11-jdk-headless
+
+# MAVEN 
+# Test installation
+mvn -version
+# Install in case it is missing
+sudo apt install maven
+```
+To prevent and identify potential permission issues on Linux machines (i.e. for executable permission), the following commands can be used to verify and manage permissions:
+
+```bash
+# Check permissions
+ls -l <REPO NAME>
+# Grant permissions
+chmod -R +rwx <REPO NAME>
+# To prevent git from identifying all files as changed (due to changed permission rights), exclude file permission (chmod) changes from git
+git config core.fileMode false
+```
+
+&nbsp;
+# 2. Using the Agent
+
+The Average Square Metre Price Estimation Agent is intended to use the `asychronous mode` of the Derivation Framework to detect changes in instantiated [OntoBuiltEnv] properties (i.e. `total floor area`, `transaction records`, and `property price index`) and automatically update associated `average price per square metre` instances in the KG. As the agent adopts `pyderivationagent`, it also serves HTTP requests to handle synchronous derivations. However, it is (strongly) discouraged to invoke such HTTP request by ONESELF. 
+
+After successful agent start-up, an instructional page shall become available at the root (i.e. `/`) of the port specified in the [docker compose file]. The exact address depends on where the agent container is deployed (i.e. localhost, remote VM, ...), but takes a form like `http://165.232.172.16:5010/`.
+
+## Asynchronous derivation operation
+Once the Agent is deployed, it periodically (every week, defined by `DERIVATION_PERIODIC_TIMESCALE`) checks the derivation that `isDerivedUsing` itself (parameter `ONTOAGENT_SERVICE_IRI`) and acts based on the status associated with that derivation. Although the [Derivation Agent] suggests the use of `.env` files to specify environment variables for agent configurations, this approach does not work properly with Docker stacks, i.e. `docker stack deploy`. Hence, the agent configuration is moved to the [docker compose file] instead.
+
+## Prior derivation markup
+
+For the Agent to detect outdated information, a proper mark up of the relevant derivation inputs (i.e. *pure* inputs) is required. (Please note, that another pre-requisite for detecting derivation inputs is the registration of the agent in the KG, i.e. `REGISTER_AGENT=true` in the [docker compose file].) The following methods from the `pyderivationagent` package shall be used to mark up derivation inputs within the KG (for illustration purposes only):
+```bash
+# Retrieve derivation client from derivation agent
+deriv_client = agent.derivationClient
+
+# Add and update time stamp to all pure input instances (i.e. inputIRIs)
+deriv_client.addTimeInstance(inputIRI)
+deriv_client.updateTimestamp(inputIRI)
```

Comment:
As we discussed, updating the timestamp here will be sufficient when your input agent updates the values.